contracts/cw-core/src/contract.rs

```diff
@@ -312,15 +317,38 @@ pub fn execute_update_proposal_modules(
     env: Env,
     sender: Addr,
     to_add: Vec<ModuleInstantiateInfo>,
-    to_remove: Vec<String>,
+    to_disable: Vec<String>,
 ) -> Result<Response, ContractError> {
     if env.contract.address != sender {
         return Err(ContractError::Unauthorized {});
     }
 
-    for addr in to_remove {
+    for addr in to_disable {
         let addr = deps.api.addr_validate(&addr)?;
-        PROPOSAL_MODULES.remove(deps.storage, addr);
+        let mut module = PROPOSAL_MODULES
+            .load(deps.storage, addr.clone())
+            .map_err(|_| ContractError::ProposalModuleDoesNotExist {
+                address: addr.clone(),
+            })?;
+
+        if module.status == ProposalModuleStatus::Disabled {
```

Comment:
nit: `!= Enabled`

Reply:
so I agree with the nit above, but by that same logic I think we may actually want to check for `== Disabled` here since we are specifically trying to disable modules

Reply:
great point!